itest/psbt_test.go

```diff
@@ -1608,6 +1619,436 @@ func testPsbtSighashNoneInvalid(t *harnessTest) {
+	// transfered assets, and one output that carries the bitcoin change.
+	require.Len(t.t, btcpsbt.Inputs, 1)
+	require.Len(t.t, btcpsbt.Outputs, 2)
+
+	// Let's set an actual address for Alice's output.
+	addrResp := t.lndHarness.Alice.RPC.NewAddress(&lnrpc.NewAddressRequest{
+		Type: lnrpc.AddressType_TAPROOT_PUBKEY,
+	})
+
+	aliceP2TR, err := btcutil.DecodeAddress(
+		addrResp.Address, harnessNetParams,
+	)
+	require.NoError(t.t, err)
+
+	alicePkScript, err := txscript.PayToAddrScript(aliceP2TR)
+	require.NoError(t.t, err)
+
+	// These are basically Alice's terms that she signs the assets over:
+	// Send me 69420 satoshis to this address that belongs to me, and you
+	// will get assets in return.
+	btcpsbt.UnsignedTx.TxOut[0].PkScript = alicePkScript
+	btcpsbt.UnsignedTx.TxOut[0].Value = 69420
+	derivation, trDerivation := getAddressBip32Derivation(
+		t.t, addrResp.Address, t.lndHarness.Alice,
+	)
+	btcpsbt.Outputs[0].Bip32Derivation = []*psbt.Bip32Derivation{
+		derivation,
+	}
+	btcpsbt.Outputs[0].TaprootBip32Derivation = []*psbt.TaprootBip32Derivation{
+		trDerivation,
+	}
+	btcpsbt.Outputs[0].TaprootInternalKey = trDerivation.XOnlyPubKey
+
+	var b bytes.Buffer
+	err = btcpsbt.Serialize(&b)
+	require.NoError(t.t, err)
+
+	// Now we need to commit the vPSBT and PSBT, creating all the related
+	// proofs for this transfer to be valid.
+	resp, err := alice.CommitVirtualPsbts(
+		ctxb, &wrpc.CommitVirtualPsbtsRequest{
+			VirtualPsbts: [][]byte{signedResp.SignedPsbt},
+			AnchorPsbt:   b.Bytes(),
+			AnchorChangeOutput: &wrpc.CommitVirtualPsbtsRequest_Add{
+				Add: true,
+			},
+			Fees: &wrpc.CommitVirtualPsbtsRequest_TargetConf{
+				TargetConf: 12,
+			},
+		},
+	)
+	require.NoError(t.t, err)
+
+	// Now we retrieve the bitcoin PSBT from the response.
+	btcpsbt, err = psbt.NewFromRawBytes(
+		bytes.NewReader(resp.AnchorPsbt), false,
+	)
+	require.NoError(t.t, err)
+
+	// The first input is the anchor that carries Alice's assets. This input
+	// will only commit to itself and Alice's output at same index (0), which
+	// is Alice's btc reward for swapping the assets. With the following sighash
+	// flag we commit to exactly that input and output, but we also allow anyone
+	// to add their own inputs, which will allow Bob later to add his btc input
+	// to pay Alice.
+	btcpsbt.Inputs[0].SighashType = txscript.SigHashSingle |
+		txscript.SigHashAnyOneCanPay
+
+	// We now strip the extra input that was only used to fund the bitcoin
+	// psbt. This is meant to be filled later by the person redeeming this
+	// swap offer.
+	btcpsbt.Inputs = append(
+		btcpsbt.Inputs[:1], btcpsbt.Inputs[2:]...,
+	)
+	btcpsbt.UnsignedTx.TxIn = append(
+		btcpsbt.UnsignedTx.TxIn[:1], btcpsbt.UnsignedTx.TxIn[2:]...,
+	)
+
+	// Let's get rid of the change output that we no longer need.
+	btcpsbt.Outputs = btcpsbt.Outputs[:2]
+	btcpsbt.UnsignedTx.TxOut = btcpsbt.UnsignedTx.TxOut[:2]
+
+	t.Logf("Funded BTC PSBT: %v", spew.Sdump(btcpsbt))
```

Comment:
Is this still a "funded" PSBT? I think we've just stripped that funding. I think at this point it's funded with the taproot-asset anchoring only, is that correct?

Reply:
Right, will rename to something like "Alice BTC PSBT"